zrml/global-disputes/src/lib.rs

```diff
@@ -272,28 +407,32 @@ mod pallet {
         ) -> DispatchResultWithPostInfo {
             ensure_signed(origin)?;
 
-            let mut winner_info =
-                <Winners<T>>::get(market_id).ok_or(Error::<T>::NoGlobalDisputeStarted)?;
-            ensure!(winner_info.is_finished, Error::<T>::UnfinishedGlobalDispute);
+            let mut gd_info = <GlobalDisputesInfo<T>>::get(market_id)
+                .ok_or(Error::<T>::NoGlobalDisputeInitialized)?;
+            ensure!(gd_info.status == GDStatus::Finished, Error::<T>::UnfinishedGlobalDispute);
 
             let winning_outcome: Option<OutcomeInfoOf<T>> =
-                <Outcomes<T>>::get(market_id, &winner_info.outcome);
+                <Outcomes<T>>::get(market_id, &gd_info.winner_outcome);
             let mut owners_len = 0u32;
-            // move the winning outcome info to Winners before it gets drained
+            // move the winning outcome info to GlobalDisputesInfo before it gets drained
             if let Some(outcome_info) = winning_outcome {
-                owners_len = outcome_info.owners.len() as u32;
+                if let Some(Possession::Shared { owners }) = &outcome_info.possession {
+                    owners_len = owners.len() as u32;
+                }
                 // storage write is needed here in case,
                 // that the first call of reward_outcome_owner doesn't reward the owners
                 // this can happen if there are more than RemoveKeysLimit keys to remove
```

Comment:
Totally understandable that nobody understands this comment. It's a bit tricky here. Maybe you have a better suggestion for the behaviour here. `purge_outcomes` removes the outcomes, but one of them contains the winner outcome owners and we would like to store those in another storage item. Because otherwise we wouldn't have saved the account id or ids which will eventually get the vote fees of the loosers in the `reward_outcome_owner` extrinsic.

Reply:
Oh I am sorry. Typical case that comments don't age well.

```rust
                // storage write is needed here in case,
                // that the first call to purge_outcomes 
                // doesn't save the owners of the winning outcome
                // saving this information is required to reward the winners
                // this can happen if there are more than RemoveKeysLimit keys to remove
```

Reply:
52925b62c3bb76ac0290609a6c71964bb799ebbb

Reply:
Ah, so by "first" you mean "any of the previous calls"?

Reply:
I meant first call, because I query the Outcomes here.

```rust
let winning_outcome: Option<OutcomeInfoOf<T>> =
                <Outcomes<T>>::get(market_id, &gd_info.winner_outcome);
```

So this returns only `Some(outcome_info)` if `Outcomes` has not already drained this `gd_info.winner_outcome`.

Reply:
I still don't get it. The comment says: "storage write is needed here in case, that the first call to purge_outcomes doesn't save the owners of the winning outcome". But when is that actually the case? As per my understand, the dispute is finished when `purge_outcomes` is allowed to be called, and during the first call of `purge_outcomes`, the winning outcome is stored. I don't see any scenario where that wouldn't be the same.

Reply:
Oh yeah. This comment is confusing. I see now. The other way around this is correct:

=> Storage write is needed in case to save the owners of the winning outcome before the draining.